### PR TITLE
Remove dangling images from GHCR on schedule

### PR DESCRIPTION
**Pull request by ferferga**
Continuation of #940

## Description
Contrary to DockerHub, GHCR keeps all the images uploaded, even those that are no longer associated to a tag. For example, updating the ``latest`` tags will keep the older image still around in the registry, but untagged.

This new workflow runs every day at 0:00 UTC and removes all the untagged images (aka dangling images)

## Motivation and Context
* There's no point in keeping untagged images at all but GitHub keeps everything by default. CI in general takes a lot of energy, storage and resources. Although those resources are free of charge for us, we don't need them and the environment will be happy with us 😊.

## How Has This Been Tested?
* Ran the script locally.
* Run this same workflow in [my personal repo](https://github.com/ferferga/ferferga)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

## Needed tokens
A secret ``ACCESS_TOKEN`` (a PAT for the org) must be added in order to remove dangling images, with the following permissions:

* ``delete:packages``
* ``read:packages``

If you prefer another name instead of ``ACCESS_TOKEN`` please let me know.

### CODE REVIEW

**Comment by TonyRL** [comment]
FYI, both actions, [GHCR Pruning](https://github.com/marketplace/actions/ghcr-pruning)  and [Container Retention Policy](https://github.com/marketplace/actions/container-retention-policy), support removing dangling images from GHCR.

**Comment by PromoFaux** [comment]
I wonder if https://github.com/pi-hole/.github might be a better place for this (or similar using one of the abovementioned actions)

**Comment by ferferga** [comment]
@PromoFaux The problem is that GH shuts down scheduled actions in a repository after 2 months with no activity, so unless that repo receives activity (not sure what GH considers activity), it won't work properly and will need constant maintenance from your side.

Here's an screenshot of the email I received in one repo where I had no activity:

File: .github/workflows/maintenance.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    # Remaining images will be removed in the next scheduled run.
+    continue-on-error: true
+    # Pay attention that the org name is not necessary here, as gh will automatically take the one from the logged in user.
+    # REMEMBER TO KEEP THIS UP TO DATE WITH A LIST OF THE PUBLISHED PACKAGES.
+    strategy:
+      fail-fast: false
+      matrix:
+        container:
+          - 'pihole'
+          - 'ftl-build'
```
Comment (PromoFaux) [inline .github/workflows/maintenance.yml:22]:
```suggestion
          - 'ftl-build'
          - 'docker-pi-hole-base'
```

File: .github/workflows/maintenance.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    steps:
+      - name: Clone repository 🔽
+        uses: actions/checkout@v2.3.4
+
+      - name: Run deletion script 🗑
+        run: ./delete_ghcr_dangling_images.sh ${{ matrix.container }}
+        working-directory: ./.github/workflows/maintenance
+        env:
+          GITHUB_TOKEN: ${{ secrets.ACCESS_TOKEN }}
```
Comment (PromoFaux) [inline .github/workflows/maintenance.yml:33]:
```suggestion
          GITHUB_TOKEN: ${{ secrets.DELETE_PACKAGE_TOKEN }}
```

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**